File: api-python/samples/server/shared.py
from fastapi import APIRouter, HTTPException, Depends, Request
from fastapi.responses import RedirectResponse
from pydantic import BaseModel
from trinsic_api.api.sessions_api import SessionsApi, GetSessionResultRequest
from trinsic_api.models import RecommendationInfo, RecommendProvidersRequest
import os
sharedRouter = APIRouter()

@sharedRouter.get("/redirect")
async def redirect(request: Request):
    return RedirectResponse(url=f"/redirect.html?{request.query_params}")

@sharedRouter.get("/providers")
async def get_providers(request: Request, sessions_api: SessionsApi = Depends()):
    ip_address = request.query_params.get("ipAddress")  # Get query param
    ip_addresses = [ip_address] if ip_address else []
    req = RecommendProvidersRequest(recommendation_info=RecommendationInfo(ip_addresses=ip_addresses), verification_profile_id=os.getenv("TRINSIC_VERIFICATION_PROFILE_ID"))
    data = sessions_api.recommend_providers(req)
    return data

class GetResultsRequest(BaseModel):
    sessionId: str
    resultsAccessKey: str

@sharedRouter.post("/exchange-result")
async def exchange_result(request: GetResultsRequest, sessions_api: SessionsApi = Depends()):
    req = GetSessionResultRequest(resultsAccessKey=request.resultsAccessKey)
    data = sessions_api.get_session_result(session_id=request.sessionId, get_session_result_request=req)

    return data

File: api-go/samples/server/shared.go
package main

import (
	"os"

	"github.com/gofiber/fiber/v2"
	trinsic_api "github.com/trinsic-id/sdk-go-api/v3"
)

func SharedRoutes(app *fiber.App, api *trinsic_api.APIClient) {
	app.Static("/", "../../../ui-web/samples/dist")

	app.Get("/redirect", func(c *fiber.Ctx) error {
		queryParams := c.OriginalURL()[len(c.Path()):]
		return c.Redirect("/redirect.html" + queryParams)
	})

	app.Get("/providers", func(c *fiber.Ctx) error {
		ipAddress := c.Query("ipAddress")
		verificationProfileId := os.Getenv("TRINSIC_VERIFICATION_PROFILE_ID")
		
		var recommendationInfo trinsic_api.NullableRecommendationInfo

		if ipAddress != "" {
			recommendationInfo = *trinsic_api.NewNullableRecommendationInfo(
				&trinsic_api.RecommendationInfo{
					IpAddresses: []string{ipAddress},
				},
			)
		}

		request := trinsic_api.RecommendProvidersRequest{
			RecommendationInfo:    recommendationInfo,
			VerificationProfileId: verificationProfileId,
		}

		req := api.SessionsAPI.RecommendProviders(c.Context()).RecommendProvidersRequest(request)

		data, _, err := req.Execute()

		if err != nil {
			return err
		}

		return c.JSON(data)
	})

	type GetResultsRequest struct {
		SessionId string `json:"sessionId"`
		AccessKey string `json:"resultsAccessKey"`
	}

	app.Post("/exchange-result", func(c *fiber.Ctx) error {
		req := new(GetResultsRequest)

		if err := c.BodyParser(req); err != nil {
			return c.Status(400).JSON(fiber.Map{
				"error": err.Error(),
			})
		}

		request := trinsic_api.GetSessionResultRequest{
			ResultsAccessKey: req.AccessKey,
		}

		data, _, err := api.SessionsAPI.GetSessionResult(c.Context(), req.SessionId).GetSessionResultRequest(request).Execute()

		if err != nil {
			return err
		}

		return c.JSON(data)
	})

}
